### PR TITLE
gh-92032: Add soft keywords to rlcompleter

### PR DESCRIPTION
**Pull request by kbeldan**
#92032

### CODE REVIEW

**Comment by kbeldan** [comment]
Hi, I figured this one would skip issue / news

**Comment by AlexWaygood** [comment]
I've added "skip issue", but if this is a bugfix that will result in a change in behaviour, I think a NEWS entry would probably be warranted :)

**Comment by AlexWaygood** [comment]
> Not a bugfix, with this the interpreter can autocomplete the match / case / _(wildcard pattern), which are as-of-now the only [soft keywords](https://docs.python.org/dev/reference/lexical_analysis.html#soft-keywords) added with [PEP-634](https://peps.python.org/pep-0634/)

In which case it is a feature, and there's even more need for it to have a news entry :)

We generally only skip news for things like docs changes, minor typo fixes in comments, changes to the CI setup, etc.. Pretty much everything else needs an entry.

**Comment by kbeldan** [comment]
To answer your request, given your repository rules, that would require a GH issue.

**Comment by kbeldan** [comment]
squashed and reworded after NEWS entry generated by blurb-it

**Comment by kbeldan** [comment]
I have rebased because upstream has evolved since, however I don't know how that affects the process.
At this stage people let their PRs lying in the pile ?

File: Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+The interpreter can now autocomplete soft keywords, as of now the PEP-634 match / case / _(wildcard pattern).
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst]:
```suggestion
The interpreter can now autocomplete soft keywords, as of now 
`match`, `case`, and `_` (wildcard pattern) from :pep:`634`.
```

**Review by JelleZijlstra — APPROVED** [review]
Tried it out locally and it works, thank you!

**Comment by JelleZijlstra** [comment]
> I have rebased because upstream has evolved since, however I don't know how that affects the process. At this stage people let their PRs lying in the pile ?

Rebasing is generally discouraged because it requires force pushing, which can make reviews harder. You can click the "update branch" button to bring the branch up to date with main.

File: Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The interpreter can now autocomplete soft keywords, as of now
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst]: Trailing whitespace

```suggestion
The interpreter can now autocomplete soft keywords, as of now
```

Comment (kbeldan) [reply Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst]: Although sphinx-lint passes locally on my side I believe it is the '_' not being escaped.

Comment (AlexWaygood) [reply Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst]: It is the trailing whitespace https://github.com/python/cpython/runs/6260946069?check_suite_focus=true
 22681C6A-6904-4186-A7B8-FABC93555344

File: Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+`match`, `case`, and `_` (wildcard pattern) from :pep:`634`.
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst]:
```suggestion
``match``, ``case``, and ``_`` (wildcard pattern) from :pep:`634`.
```

File: Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst (hunk 2 above)
Comment (JelleZijlstra) [reply Misc/NEWS.d/next/Core and Builtins/2022-04-28-20-19-49.gh-issue-92032.ef-UfM.rst]: Thanks, I think both of these needed fixing. I was still in `md` mode when writing that.

**Comment by kbeldan** [comment]
Last test is stuck
stuck_in_concurrent_futures

**Comment by JelleZijlstra** [comment]
> Last test is stuck stuck_in_concurrent_futures

I turned it off and on again

**Comment by AlexWaygood** [comment]
Congrats on your first CPython contribution, @kbeldan! 😃

**Comment by AlexWaygood** [comment]
> Last test is stuck stuck_in_concurrent_futures

I've opened #92222, since this is also happening on my completely unrelated PR here: https://github.com/python/cpython/runs/6270908903?check_suite_focus=true

**Pull request merged**